server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -142,17 +141,14 @@ public RemoteRestoreResult restore(
     ) {
         Map<String, Tuple<Boolean, IndexMetadata>> indexMetadataMap = new HashMap<>();
         ClusterState remoteState = null;
-        boolean metadataFromRemoteStore = (restoreClusterUUID == null
-            || restoreClusterUUID.isEmpty()
-            || restoreClusterUUID.isBlank()) == false;
+        boolean metadataFromRemoteStore = restoreClusterUUID != null && !restoreClusterUUID.isEmpty();
```

Comment:
This is not logically equivalent as isBlank will check for whitespace.  Why do we need to update this check?

Reply:
I see, I dint expect we can get the restoreClusterUUID as whitespace and mess the boolean, while understanding the logic I modified this to do some clean up. But seeing your comment I will revert to default.
```
        boolean metadataFromRemoteStore = (restoreClusterUUID == null
            || restoreClusterUUID.isEmpty()
            || restoreClusterUUID.isBlank()) == false;
 ```

Reply:
yes pls :) This change is already really large please only change whats needed.